plugins/modules/rds_instance_param_group_info.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024 Ansible Project
+# Copyright (c) 2024 Mandar Vijay Kulkarni (@mandar242)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+module: rds_instance_param_group_info
+version_added: 9.1.0
+short_description: Describes the RDS parameter group.
+description:
+  - Describe a specific RDS parameter group, the parameter group associated with a specified RDS instance, or all parameter groups available in the current region.
+options:
+  db_instance_identifier:
+    description:
+      - The RDS instance's unique identifier.
+    required: false
+    type: str
+  db_parameter_group_name:
+    description:
+      - The name of a specific DB parameter group to return details for.
+    required: false
+    type: str
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+"""
+
+EXAMPLES = r"""
+- name: Get specific DB instance's parameter group info
+  amazon.aws.rds_instance_param_group_info:
+    db_instance_identifier: database-1
+
+- name: Get specific DB parameter group's info
+  amazon.aws.rds_instance_param_group_info:
+    db_parameter_group_name: my-test-pg
+
+- name: Get all parameter group info from the region
+  amazon.aws.rds_instance_param_group_info:
+"""
+
+RETURN = r"""
+db_parameter_groups:
+  description: List of RDS instance parameter groups.
+  returned: always
+  type: list
+  contains:
+    db_parameter_group_name:
+      description:
+        - The name of the RDS instance parameter group.
+      type: str
+    db_parameter_group_family:
+      description:
+        - The name of the RDS parameter group family that this RDS instance parameter group is compatible with.
+      type: str
+    description:
+      description:
+        - Provides the customer-specified description for this RDS instance parameter group.
+      type: str
+    db_parameter_group_arn:
+      description:
+        - The Amazon Resource Name (ARN) for the RDS instance parameter group.
+      type: str
+"""
+
+from typing import Any, List
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+from ansible_collections.amazon.aws.plugins.module_utils.modules import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.rds import describe_db_instances
+from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
+
+
+def get_db_instance_param_group_name(connection: Any, module: AnsibleAWSModule) -> str:
```

Comment:
It appears that this function is already addressed by [rds_instance_info.py](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance_info.py#L369). Do we really need it here? While it does offer additional functionality, I believe it may be redundant in order to align with the rds_cluster_param_group_info module.

Reply:
It is redundant however I think it would be good to have that function in the code as it allows below task
First code finds the name of parameter group attached to the rds instance, and uses it to find that param group's info
```
    - name: Get specific DB instance's parameter group info
      amazon.aws.rds_instance_param_group_info:
        db_instance_identifier: database-1
```

above can be surely achieved with combination of existing [rds_instance_info.py](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance_info.py#L369) and this new module but reduces user experience in my opinion.
```
    - name: Get specific DB instance's parameter group info
      amazon.aws.rds_instance_info:
        db_instance_identifier: database-1
      register: instance_info

    - name: Get specific DB parameter group's info
      amazon.aws.rds_instance_param_group_info:
        db_parameter_group_name: "{{ instance_info.instances[0].db_parameter_groups.db_parameter_group_name }}"
```

I'm open with going with either approach. @GomathiselviS

Reply:
removed.